macros/edr/tests/test_utils/get_compiled_test_id.sql

```diff
@@ -0,0 +1,6 @@
+{% macro get_compiled_test_id() %}
+    {% set test_node = model %}
+    {% set test_hash = test_node.unique_id.split(".")[-1] %}
+    {% set test_name = test_node.name %}
+    {% do return("{}_{}".format(test_name, test_hash)) %}
```

Comment:
Can add context on why this was needed?
I'm slightly worried about adding another type of ID for tests to our code, it's confusing :)

Reply:
The problem is conflicting names for Elementary's test tables.
The current table names are `{flattened_test.name}__metrics` or `{flattened_test.name}__anomaly_scores`.
Here's an example of the problem with the current situation:
```py
ipdb> t1 = ctx['graph']['nodes']['test.elementary_integration_tests.elementary_source_freshness_anomalies_training_any_type_column_anoma
lies_training_.5733415eda']
ipdb> t2 = ctx['graph']['nodes']['test.elementary_integration_tests.elementary_source_freshness_anomalies_training_string_column_anomali
es_training_.4640cc98af']
ipdb> t1['name']
'elementary_source_freshness_anomalies_training_any_type_column_anomalies_training_'
ipdb> t2['name']
'elementary_source_freshness_anomalies_training_string_column_anomalies_training_'
ipdb> prefix = '__anomaly_scores'
ipdb> max_length = 63
ipdb> t1["name"][:max_length - len(prefix)] + prefix
'elementary_source_freshness_anomalies_training___anomaly_scores'
ipdb> t2["name"][:max_length - len(prefix)] + prefix
'elementary_source_freshness_anomalies_training___anomaly_scores'
```

As you can see, two different tests (`t1`, `t2`) yield the same table name and therefore override each other and error upon running `dbt test`.

The solution is to include the test's hash into the table so that it's unique.
In addition to that, the table's name is now truncated from the end rather than from the start in order to include the hash. Alternatively we could've placed the hash first but then we also need to add dummy characters because the hash might start with digits, but table names need to start with letters.

Please let me know what you think.
Thanks.

Reply:
Hmm I see, makes sense
Any chance you can give it a more direct name like "get_elementary_test_table_name"? (and avoid using it in logs etc)
Is it needed anywhere else?

Reply:
Also - one problem I see with truncating from the end is that you might have a problematic character in the middle of a test name (e.g. a digit)
So you may need to add something at the beginning.

Or alternatively - truncate from the beginning but make sure the hash is added anyway.

Reply:
Thanks, that's great feedback.
I changed the implementation [here](https://github.com/elementary-data/dbt-data-reliability/pull/230/commits/90500cc2bc4c7a6bc3eb026dcdb3065e427d8972).
One thing that I'm not too crazy about is that singular tests don't have a hash at their unique ID.
Generally it doesn't matter because this macro is only called on Elementary's generic tests, but I wonder if we can find a different and more stable identifier for a test that is short. Ideally I'd hash the `unique_id` but I can't because `local_md5` is only introduced in v1.4.0. This might be good enough though.